src/python/pants/engine/goal.py

```diff
@@ -132,6 +133,11 @@ def _selects_environments(cls) -> bool:
     def name(cls) -> str:
         return cast(str, cls.subsystem_cls.name)
 
+    @final
+    @classmethod
+    def is_goal(cls) -> bool:
+        return True
```

Comment:
This should probably use `is_subclass` instead rather than a new property.

Reply:
Agree, I can take another look at that. I tried that first and failed, but I've learned so much each day, so I may be able to pull it off this time. Perhaps :P